addons/udes_stock/models/stock_move_line.py

```diff
@@ -0,0 +1,195 @@
+            # So we have a left_ml and right ml, where for qty_done the new ml is the right,
+            # else the new ml is the left one
+            # <--------------------x-------------> x = split point
+            # <----ml_left---------x--ml_right--->
+            # <----qty-------------x-------------> If qty_done == 0
+            # <----qty-done---p----x---new_ml----> If p=qty_done > 0, and x = qty > 0, split at p ignore x
+
+            left_ml_qty = split_qty
+            right_ml_qty = float_round(
+                self.product_uom_qty - split_qty,
+                precision_rounding=self.product_uom_id.rounding,
+                rounding_method="UP",
+            )
+
+            # Now compute the left/right ml_ordered_qty
+            # Note in qty_done split they just set right_ml_ordered_qty = right_ml_qty
+            left_ml_ordered_qty = split_qty
+            right_ml_ordered_qty = float_round(
+                self.ordered_qty - split_qty,
+                precision_rounding=self.product_uom_id.rounding,
+                rounding_method="UP",
+            )
+
+            if qty_split > self.ordered_qty:
+                # TODO: Why does the split by qty > 0 assign ordered_qty ot the right?
+                # Would have thought that it would be
+                # left_ml_ordered_qty = self.ordered_qty
+                # right_ml_ordered_qty = 0
+                # Is it because return the left ml, so bias orders there for split by qty?
+                left_ml_ordered_qty = self.ordered_qty if qty_done > 0 else 0
+                right_ml_ordered_qty = 0 if qty_done > 0 else self.ordered_qty
+
+            # create new move line
+            if qty_done > 0:
+                # create ml based on qty_done
+                new_ml = self.copy(
+                    default={
+                        "product_uom_qty": right_ml_qty,
+                        "ordered_qty": right_ml_ordered_qty,
+                        "qty_done": 0.0,
+                        "result_package_id": False,
+                        "lot_name": False,
+                    }
+                )
+            else:
+                new_ml = self.copy(
+                    default={
+                        "product_uom_qty": left_ml_qty,
+                        "ordered_qty": left_ml_ordered_qty,
+                        "qty_done": 0.0,
+                        "result_package_id": False,
+                        "lot_name": False,
+                    }
+                )
+            # updated ordered_qty otherwise odoo will use product_uom_qty
+            # new_ml.ordered_qty = ordered_quantity_left_todo
+            # update self move line quantity to do
+            # - bypass_reservation_update:
+            #   avoids to execute code specific for Odoo UI at stock.move.line.write()
+            self.with_context(bypass_reservation_update=True).write(
+                {"product_uom_qty": left_ml_qty if qty_done > 0 else right_ml_qty,
```

Comment:
Can we move the opening and close of the dict to the lines above and below please?
```
write({
...
})
```